hazelcast-client-legacy/src/main/java/com/hazelcast/client/spi/impl/listener/ClientEventRegistration.java

```diff
@@ -67,5 +67,27 @@ public int getCallId() {
         return callId;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        ClientEventRegistration that = (ClientEventRegistration) o;
+
+        return !(serverRegistrationId != null
```

Comment:
Can serverRegistrationId ever be not null? If it can't be null, add the null check here and add a null check using Preconditions in the constructor.